File: 70. Climbing Stairs/70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n

        num1 = 1
        num2 = 2

        for _ in range(3, n + 1):
            num1, num2 = num2, num1 + num2

        return num2

File: 70. Climbing Stairs/70. Climbing Stairs.java
class Solution {

  public int climbStairs(int n) {
    if (n <= 2) {
      return n;
    }

    int num1 = 1;
    int num2 = 2;

    for (int i = 3; i < n + 1; i++) {
      int temp = num2;
      num2 = num1 + num2;
      num1 = temp;
    }

    return num2;
  }
}
